distributed/worker.py

```diff
@@ -1219,7 +1220,7 @@ async def heartbeat(self):
                 logger.error(
                     f"Scheduler was unaware of this worker {self.address!r}. Shutting down."
                 )
-                await self.close(report=False)
+                await self.close()
```

Comment:
This and the following `self.close()` and `self.close()` in `handle_scheduler` effectively do https://github.com/dask/distributed/issues/6387. I wanted to do that a bit more carefully and add tests.